### PR TITLE
Fix: [SeamlessM4T - S2TT] Bug in batch loading of audio in torch.Tensor format in the SeamlessM4TFeatureExtractor class

### PR DESCRIPTION
**Pull request by nicholasneo78**
# What does this PR do?

Fixes # (issue)
Based on the documentation for the [SeamlessM4TProcessor](https://github.com/huggingface/transformers/blob/main/src/transformers/models/seamless_m4t/processing_seamless_m4t.py#L22) class, the class is supposed to take in either `List[np.ndarray]` or `List[torch.Tensor]` for batch decoding when `processor(audios=...)` is called. It would then return a batch of transcriptions (S2TT task of SeamlessM4T). However, when `List[torch.Tensor]` is passed into the `audios` arg, only one translated transcript is being returned even though a batch of audio is passed in. After adding the check for `torch.Tensor` in the `SeamlessM4TFeatureExtractor` class, the translated batch transcript returned as expected.

Below is a code snippet that I use to test the issue:
```python
from transformers import SeamlessM4Tv2Model, SeamlessM4TProcessor
import torch
from datasets import load_dataset

processor = SeamlessM4TProcessor.from_pretrained("facebook/seamless-m4t-v2-large")
model = SeamlessM4Tv2Model.from_pretrained("facebook/seamless-m4t-v2-large", use_safetensors=True)
dataset = load_dataset("arabic_speech_corpus", split="test", streaming=True)

# numpy array as audio_inputs
audio_sample = next(iter(dataset))["audio"]
print(type(audio_sample["array"])) # <class 'numpy.ndarray'>
# get a list of two numpy arrays to simulate batch size=2 when loading the audio arrays
audio_sample_batch = [audio_sample["array"], audio_sample["array"]]

audio_inputs = processor(audios=audio_sample_batch, return_tensors="pt", sampling_rate=16000)
output_tokens = model.generate(**audio_inputs, tgt_lang="eng", generate_speech=False)
translated_text_from_audio = processor.batch_decode(output_tokens[0].tolist(), skip_special_tokens=True)
print(f"Translated text from audio (numpy array): {translated_text_from_audio}\n")
# >>> Translated text from audio (numpy array): ['The first is the fact that the sun is shining brightly on the moon.', 'The first is the fact that the sun is shining brightly on the moon.']

# torch tensors as audio_inputs
torch_tensor_audio_sample = torch.from_numpy(audio_sample["array"])
print(type(torch_tensor_audio_sample)) # <class 'torch.Tensor'>
# get a list of two torch tensors to simulate batch size=2 when loading the audio arrays
torch_tensor_audio_sample_batch = [torch_tensor_audio_sample,torch_tensor_audio_sample]

audio_inputs = processor(audios=torch_tensor_audio_sample_batch, return_tensors="pt", sampling_rate=16000)
output_tokens = model.generate(**audio_inputs, tgt_lang="eng", generate_speech=False)
translated_text_from_audio = processor.batch_decode(output_tokens[0].tolist(), skip_special_tokens=True)
print(f"Translated text from audio (torch tensors): {translated_text_from_audio}")
# >>> Translated text from audio (torch tensors): ['The first is the fact that the sun is shining brightly on the moon.']
# expects two translated sentences just like the numpy array inputs but only one sentence is translated 
``` 

Environment:
```shell
- `transformers` version: 4.36.0.dev0
- Platform: Linux-6.1.11-76060111-generic-x86_64-with-glibc2.35
- Python version: 3.10.6
- Huggingface_hub version: 0.19.4
- Safetensors version: 0.4.1
- Accelerate version: not installed
- Accelerate config: not found
- PyTorch version (GPU?): 2.1.1+cu121 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: No
```

## Before submitting
- [ x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests), Pull Request section?

cc: @sanchit-gandhi

### CODE REVIEW

**Comment by ArthurZucker** [comment]
cc @ylacombe

File: src/transformers/models/seamless_m4t/feature_extraction_seamless_m4t.py [hunk 1]
```diff
@@ -19,6 +19,7 @@
 from typing import List, Optional, Union
 
 import numpy as np
+import torch
```
Comment (ylacombe) [inline src/transformers/models/seamless_m4t/feature_extraction_seamless_m4t.py:22]:
```suggestion
from ...utils import is_torch_available

if is_torch_available():
    import torch
```

**Review by ylacombe — COMMENTED** [review]
Thanks for flagging this @nicholasneo78, there's indeed an error in the documentation. 

Your fix should work but adds a `torch` import to the FeatureExtractor class, so it won't fit to transformers right away as the FeatureExtractor is framework independent here. That's why I suggested some changes.

Don't forget to use `make style` (see [here](https://huggingface.co/docs/transformers/contributing) for more info), to format everything to transformers standard!

Last comments from me, could you update the docstring of the FeatureExtractor to add torch tensor compatibility, and add a test in `test_feature_extraction_seamless_m4t.py`?

Let me know if this works for you! And if you don't have the bandwidth to finalize the PR, no worries, just let me know!

**Comment by nicholasneo78** [comment]
Hi @ylacombe,

Yes this works for me. Have updated the code `feature_extraction_seamless_m4t.py` and added test in `test_feature_extraction_seamless_m4t.py`

Thanks for the suggestion!

**Review by ylacombe — APPROVED** [review]
This looks great to me, thanks for taking the time to iterate here!

Let's request a core maintainer review!

File: tests/models/seamless_m4t/test_feature_extraction_seamless_m4t.py [hunk 2]
```diff
@@ -139,37 +139,59 @@ def test_feat_extract_to_json_file(self):
         dict_second = feat_extract_second.to_dict()
         self.assertEqual(dict_first, dict_second)
 
+    @require_torch
     def test_call(self):
```
Comment (amyeroberts) [inline tests/models/seamless_m4t/test_feature_extraction_seamless_m4t.py:143]: Let's do two separate test methods instead - `test_call_numpy` and `test_call_torch` - as the PT lines are all independent of the numpy ones

**Review by amyeroberts — APPROVED** [review]
Thanks for fixing! 

Just a comment on the test structure

File: tests/models/seamless_m4t/test_feature_extraction_seamless_m4t.py (hunk 2 above)
Comment (nicholasneo78) [reply tests/models/seamless_m4t/test_feature_extraction_seamless_m4t.py]: Made changes to separate the test methods. Thanks!

**Review by amyeroberts — APPROVED** [review]
Great - thanks for iterating!

**Pull request merged**